sw/airborne/modules/lidar/tfmini.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2019 Freek van Tienen <freek.v.tienen@gmail.com>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+/** @file modules/lidar/tfmini.h
+ *  @brief driver for the TFMini lidar
+ *
+ */
+#ifndef LIDAR_TFMINI_H
+#define LIDAR_TFMINI_H
+
+#include "std.h"
+#include "mcu_periph/i2c.h"
+
+enum TFMiniParseStatus {
+  TFMINI_INITIALIZE,
+	TFMINI_PARSE_HEAD,
+  TFMINI_PARSE_HEAD2,
+	TFMINI_PARSE_DIST_L,
+	TFMINI_PARSE_DIST_H,
+	TFMINI_PARSE_STRENGTH_L,
+  TFMINI_PARSE_STRENGTH_H,
+  TFMINI_PARSE_MODE,
+  TFMINI_PARSE_BYTE7,
+  TFMINI_PARSE_CHECKSUM
+};
+
+struct TFMini
+{
+  struct link_device *device;
```

Comment:
if the type of device is always the same (e.g. UART) it is probably not needed to use the abstract link_device structure

Reply:
There is also an I2C version, so I think it is nice to keep for the future